modules/packages/arch/metadata.go

```diff
@@ -0,0 +1,255 @@
+// Copyright 2023 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package arch
+
+import (
+	"crypto/md5"
+	"crypto/sha256"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"path"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/DataDog/zstd"
+	"github.com/mholt/archiver/v3"
+)
+
+// Metadata for arch package.
+type Metadata struct {
+	Filename          string
+	Name              string
+	Base              string
+	Version           string
+	Description       string
+	CompressedSize    int64
+	CompressedSizeMib string
+	InstalledSize     int64
+	InstalledSizeMib  string
+	MD5               string
+	SHA256            string
+	URL               string
+	BuildDate         int64
+	BuildDateStr      string
+	BaseDomain        string
+	Packager          string
+	Provides          []string
+	License           []string
+	Arch              []string
+	Depends           []string
+	OptDepends        []string
+	MakeDepends       []string
+	CheckDepends      []string
+	Backup            []string
+}
+
+// Function that recieves arch package archive data and returns it's metadata.
+func EjectMetadata(filename, domain string, pkg []byte) (*Metadata, error) {
+	d, err := zstd.Decompress(nil, pkg)
```

Comment:
Thanks for this PR! I haven't gone through it fully yet, but just one thing off the top. Could you use `archiver` for this instead, as I don't believe the datadog import will work on Windows builds.